pkg/autodiscovery/common/utils/prometheus.go

```diff
@@ -21,14 +22,29 @@ const (
 // buildInstances generates check config instances based on the Prometheus config and the object annotations
 // The second returned value is true if more than one instance is found
 func buildInstances(pc *types.PrometheusCheck, annotations map[string]string, namespacedName string) ([]integration.Data, bool) {
+	openmetricsVersion := config.Datadog.GetInt("prometheus_scrape.version")
+
 	instances := []integration.Data{}
 	for k, v := range pc.AD.KubeAnnotations.Incl {
 		if annotations[k] == v {
 			log.Debugf("'%s' matched the annotation '%s=%s' to schedule an openmetrics check", namespacedName, k, v)
 			for _, instance := range pc.Instances {
 				instanceValues := *instance
-				if instanceValues.URL == "" {
-					instanceValues.URL = types.BuildURL(annotations)
+				if instanceValues.PrometheusURL == "" && instanceValues.OpenMetricsEndpoint == "" {
+					switch openmetricsVersion {
+					case 1:
+						instanceValues.PrometheusURL = types.BuildURL(annotations)
+						if len(instanceValues.Metrics) == len(types.DefaultPrometheusCheck.Instances[0].Metrics) &&
+							&instanceValues.Metrics[0] == &types.DefaultPrometheusCheck.Instances[0].Metrics[0] {
+							instanceValues.Metrics = types.OpenmetricsDefaultMetricsV1
+						}
```

Comment:
Nit: this logic is duplicated in `case 2`, should we have it in a separate function that takes `types.OpenmetricsDefaultMetricsV1` or `types.OpenmetricsDefaultMetricsV2` as argument depending on the case ? Feel free to ignore if you find it useless.